app/models/host/hostmix.rb

```diff
@@ -5,6 +5,10 @@ def has_many_hosts(options = {})
     end
 
     def belongs_to_host(options = {})
+      belongs_to :host, {:class_name => "Host::Base", :foreign_key => :host_id}.merge(options)
```

Comment:
Might not this change break plugins using this method? maybe keep the same method but make `:class_name` a kwarg that defaults to `Host::Managed`, to more clearly indicate this can be overridden when needed?